I found low-coverage Whole Genome Sequencing data by:

*1)* Searching for "human low-coverage Whole Genome Sequencing" in the [SRA](https://www.ncbi.nlm.nih.gov/sra/?term=human+low-coverage+whole+genome+sequencing), and then filtering for runs with **.bam** files available
*2)* I then clicked "Send results to Run selector" to view those results in the [SRA Run Selector](https://www.ncbi.nlm.nih.gov/Traces/study/?)
*3)* I exported the meta data, and I then filtered for runs with 100Mb to 500Mb (and then for the 200 smallest .bam files, for running on my local computer)
*4)* With the run numbers (and knowledge that .bam files existed), I downloaded .bam files for alignments to all chromosmes from the [ENA](https://www.ebi.ac.uk/ena)
