I first tested a very simple strategy (looking for the presence of a variant or a reference match in at least one read, and defining the variant as heterozygous if it was present).  However, I couldn't do this for the Color lcWGS data (for the subset of 1000 Genomes variants that I was checking), and this showed a noticable loss in the relatedness estimates between my re-aligned Nebula lcWGS and SNP chip samples (producing kinship estimates of ~0.24, instead of a kinship estimate of ~0.50 between my 23and Me and Genes for Good SNP chips).

**So, I thought I needed to instead see what I could accomplish with some imputation strategies.**  Since the strategy above wasn't very good at defining a starting set of genotypes, I needed to find strategies that start with FASTQ or BAM input files (and/or provide an alternative way to start with a small set of more accurate genotypes, upstream of imputation).  For example, you can see much better accuracy for my [imputed Nebula lcWGS genotypes](https://github.com/cwarden45/DTC_Scripts/blob/master/Nebula/Downsample_IBD/README.md), although that was also unacceptable in some [other ways](http://cdwscience.blogspot.com/2019/08/low-coverage-sequencing-is-not.html).
